File: HELLO_WEB/src/main/java/kr/or/aiai/dao/MyUpdate.java
package kr.or.aiai.dao;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class MyUpdate {
	public static void main(String[] args) {
		Connection conn = null;
		PreparedStatement pstmt = null;

		try {
			
			//String driver = "org.mariadb.jdbc.Driver";
			String driver = "com.mysql.cj.jdbc.Driver";
			
			//String url = "jdbc:mariadb://localhost:3305/python";
			String url = "jdbc:mysql://localhost:3305/python";
			String user = "root";
			String pw = "python";
			
			Class.forName(driver);
			
			conn = DriverManager.getConnection(url, user, pw);
			
			String sql = "Update emp set e_name=?, sex=?, addr=? where e_id=?";
			
			pstmt = conn.prepareStatement(sql);
			pstmt.setString(1, "4");
			pstmt.setString(2, "4");
			pstmt.setString(3, "4");
			pstmt.setInt(4, 3);
			
			int cnt = pstmt.executeUpdate();
			if(cnt > 0) {
				System.out.println(cnt + "개의 데이터가 수정되었습니다");				
			}else {
				System.out.println("데이터 추가 실패");
			}

			
		} catch (SQLException | ClassNotFoundException e) {
			e.printStackTrace();
		} finally {
			try {pstmt.close();} catch (SQLException e) {e.printStackTrace();}
			try {conn.close();} catch (SQLException e) {e.printStackTrace();}	
		}
	}
}

File: Hello_PYTHON/day07/myupdate.py
import pymysql

conn = pymysql.connect(host='localhost', port=3305,
                       user='root', password='python',
                       db='python', charset='utf8')

cur = conn.cursor() #jdbc에서 statement역할

# python 3.5 이상부터는 fString을 쓸 수 있다.

e_id = "3"
e_name = "4"
sex = "4"
addr = "4"

sql = f"""
update emp 
   set e_name='{e_name}', sex='{sex}', addr='{addr}'
 where e_id='{e_id}'
"""

cnt= cur.execute(sql)

conn.commit()
print(cnt, "개의 레코드가 수정되었습니다.")

cur.close()
conn.close()